pkg/maintainer/spv/spv.go

```diff
@@ -49,18 +57,384 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
+		len(depositSweepTransactions),
+	)
+
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get unproven deposit sweep transactions: [%v]",
+			err,
+		)
+	}
+
+	txProofDifficultyFactor, err := sm.spvChain.TxProofDifficultyFactor()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get transaction proof difficulty factor: [%v]",
+			err,
+		)
+	}
+
+	// TODO: Handle a situation in which the block headers in the proof span
+	//       multiple Bitcoin difficulty epochs.
+	requiredConfirmations := uint(txProofDifficultyFactor.Uint64())
+
+	currentEpochDifficulty, previousEpochDifficulty, err :=
+		sm.btcDiffChain.GetCurrentAndPrevEpochDifficulty()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get Bitcoin epoch difficulties: [%v]",
+			err,
+		)
+	}
+
+	for _, transaction := range depositSweepTransactions {
+		// Print the transaction in the same endianness as block explorers do.
+		transactionHashStr := transaction.Hash().Hex(bitcoin.ReversedByteOrder)
+
+		logger.Infof(
+			"Proceeding with deposit sweep proof for transaction [%s]",
+			transactionHashStr,
+		)
+
+		accumulatedConfirmations, err := sm.btcChain.GetTransactionConfirmations(
+			transaction.Hash(),
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get transaction confirmations: [%v]",
+				err,
+			)
+		}
 
-	// TODO: Implement. For now, just wait.
-	<-ctx.Done()
+		if accumulatedConfirmations < requiredConfirmations {
+			// Skip the transaction as it has not accumulated enough
+			// confirmations. It will be proven later.
+			logger.Infof(
+				"Skipped proving deposit sweep transaction [%s] due to "+
+					"transaction not having enough confirmations yet",
+				transactionHashStr,
+			)
+			continue
+		}
+
+		_, proof, err := bitcoin.AssembleSpvProof(
+			transaction.Hash(),
+			requiredConfirmations,
+			sm.btcChain,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to assemble SPV proof: [%v]", err)
+		}
+
+		firstBlockHeaderDifficulty := proof.FirstBlockHeaderDifficulty()
+
+		if firstBlockHeaderDifficulty.Cmp(currentEpochDifficulty) != 0 &&
+			firstBlockHeaderDifficulty.Cmp(previousEpochDifficulty) != 0 {
+			// Skip the transaction as the difficulty of the first block
+			// header in the proof does not match the current or the
+			// previous difficulty seen on-chain. The reason for this is
+			// most likely that the proof consists of block headers from the
+			// beginning of a new Bitcoin difficulty epoch and the epoch is
+			// not proven on-chain yet. In that case the transaction will be
+			// proven in the future. The other reason could be that the
+			// transaction is older than the last two Bitcoin difficulty
+			// epoch. In that case the transaction will soon leave the
+			// sliding window of recent transactions.
+			logger.Warnf(
+				"Skipped proving deposit sweep transaction [%s] due to "+
+					"difficulties mismatch between proof and relay",
+				transactionHashStr,
+			)
```

Comment:
Let's be more specific:

```suggestion
			logger.Warnf(
				"skipped proving deposit sweep transaction [%s]; found "+
					"difficulties mismatch between proof and relay: "+
					"proof first header difficulty: [%v], "+
					"previous relay difficulty: [%v], "+
					"current relay difficultry: [%v]",
				transactionHashStr,
				firstBlockHeaderDifficulty,
				previousDifficulty,
				currentDifficulty,
			)
```

Reply:
Done in 69922dff13522866329c1ca88f804e025ecefb2c.